### PR TITLE
fix: refine reconcile by run dry-run update for filling default values

### LINKED ISSUE #301: [Bug]: Cannot complete reconcile if `overlay` does not include fields that have default value
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

- Version or commit-id (e.g. v0.1.0 or 8b23a93): v0.6.0

### Actual Behavior

Set the following overlay will cause controller cannot complete reconcile:

```
    overlay:
      volumes:
      - name: cert
        secret:
          secretName: sql-cert
```

The reason is that k8s will set defaultMode for `volume`, the volume in podTemplate would be like:

```
      volumes:
      - name: cert
        secret:
          secretName: sql-cert
          defaultMode: 420
```

Thus the desired spec will always differ with the computed spec.

### Expected Behavior

The default value set by k8s should be ignore when compare diffs

### Steps to Reproduce

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by loveRhythm1990**
**What type of PR is this?**

- [ ] API-change
- [x] BUG
- [ ] Improvement
- [ ] Documentation
- [ ] Feature
- [ ] Test and CI
- [ ] Code Refactoring

**Which issue(s) this PR fixes:**

Fixes #301

**What this PR does / why we need it:**

Controller reconcile cannot stop if applied overly configs have default values in kube-apiserver side, this PR eliminates the difference between we applied spec and kube-apiserver side spec, by run dry-run update to fill default values

**Special notes for your reviewer:**

Not Available

**Additional documentation (e.g. design docs, usage docs, etc.):**

Not Available

### CODE REVIEW

**Review by wanglei4687 — APPROVED** [review]
LGTM🤣

File: api/core/v1alpha1/common_helpers.go [hunk 1]
```diff
@@ -100,9 +100,6 @@ func (o *Overlay) OverlayPodSpec(pod *corev1.PodSpec) {
 	if o.ImagePullSecrets != nil {
 		pod.ImagePullSecrets = o.ImagePullSecrets
 	}
-	if o.Affinity != nil {
```
Comment (aylei) [inline api/core/v1alpha1/common_helpers.go:103]: Why do we need to remove this?

File: pkg/controllers/logset/controller.go [hunk 2]
```diff
@@ -113,6 +113,11 @@ func (r *Actor) Observe(ctx *recon.Context[*v1alpha1.LogSet]) (recon.Action[*v1a
 	if err := syncPods(ctx, sts); err != nil {
 		return nil, err
 	}
+
+	// let apiserver fill default field values for us by dry-run, otherwise following 'Semantic.DeepEqual' may always be false
+	if err = ctx.Update(sts, client.DryRunAll); err != nil {
```
Comment (aylei) [inline pkg/controllers/logset/controller.go:118]: Could you add a test case to cover this fix?

**Review by aylei — COMMENTED** [review]
the rest LGTM

**Pull request merged**